File: GeneticControllerOptimiser/GeneticControllerOptimiser/Classes/SystemState.cs
namespace GeneticControllerOptimiser.Classes
{
    using global::System;

    class SystemState
    {
        public double X { get; set; }
        public double Y { get; set; }
        public double DxDt { get; set; }
        public double DyDt { get; set; }
        public double Angle { get; set; }
        public double DAngleDt { get; set; }

        public SystemState()
        {
            Angle = Math.PI;
        }

        public override string ToString()
        {
            return string.Format("X {0} Y {1} Angle {2}", X, Y, Angle);
        }
    }
}

File: GeneticControllerOptimiser/FSharpVersion/SystemState.fs
module SystemState

type ThrusterState = {
    Up : float
    Left : float
    Right : float
    }

let LimitThrusterState state = 
    let limitValue lower upper = function
        | n when n <= lower -> lower
        | n when n >= upper -> upper
        | n -> n

    {
        Up = limitValue 0.0 1.0 state.Up
        Left = limitValue 0.0 1.0 state.Left
        Right = limitValue 0.0 1.0 state.Right
    }

type SystemState = {
        X : float
        Y : float
        DxDt : float
        DyDt : float
        Angle : float
        DAngleDt : float
    }